lua/ttt2/extensions/draw.lua

```diff
@@ -457,6 +457,54 @@ function draw.AdvancedText(text, font, x, y, color, xalign, yalign, shadow, scal
 	end
 end
 
+-- If there are no spaces, we have to cut the string at some point.
+-- To improve performance, we don't want to iterate over every single
+-- character. Therefore we assume the length based on an average first.
+local function InternalSplitLongWord(word, width, widthWord)
+	local charCount = utf8.len(word)
+	local wCharAverage = widthWord / charCount
+	local charCountPerLine = math.floor(width / wCharAverage * 0.8) -- decreae a bit to have tolerance
+	local lastStartPos = 1
+	local lineNumber = 1
+
+	local lines = {""}
+
+	while true do
+		local nextStartPos = lastStartPos + charCountPerLine
+		lines[lineNumber] = utf8.sub(word, lastStartPos, nextStartPos -1)
+		lastStartPos = nextStartPos
+
+		local wLine
+
+		-- then iterate over further chars until line end is reached
+		for i = lastStartPos, charCount do
+			local added = lines[lineNumber] .. utf8.GetChar(word, i)
+
+			wLine = surface.GetTextSize(added)
+
+			if wLine > width then
```

Comment:
Then I'd have to check in both direction which would make this whole thing more comlex

Reply:
no it makes it easier.

You just assume the maximum amount of characters that can fit based on the average. 
If good -> fine
If too big -> just go 1 by 1 and remove a character until it fits.

Reply:
I get what you're trying to say, but the easiest solution is to keep it as it is right now. Because it is already working. I don't have the motivation to rework this a second time

Reply:
i can do it if you do not want to